src/Authentication/Authentication/Helpers/EncodeUriPath.cs

```diff
@@ -18,12 +18,21 @@ public static Uri EscapeDataStrings(this Uri uri)
             int counter = 0;
             var pathSegments = uri.OriginalString.Split('/');
             StringBuilder sb = new StringBuilder();
-            foreach (var segment in pathSegments)
+            foreach (var s in pathSegments)
             {
+                var segment = s;
                 //Skips the left part of the uri i.e https://graph.microsoft.com
                 if (counter > 2)
                 {
                     sb.Append('/');
+                    if(s.Contains("?"))
+                    {
+                        var queryStringIndex = segment.IndexOf("?");
+                        string queryString = segment.Substring(queryStringIndex);
+                        segment = s.Substring(0, queryStringIndex);
+                        sb.Append(Uri.EscapeDataString(segment));
+                        sb.Append(queryString);
+                    }
```

Comment:
Do you need an `else` block here? Looks  the appending will happen twice?

Reply:
Yes added